hydra/core/singleton.py

```diff
@@ -18,6 +18,12 @@ def instance(cls: Any, *args: Any, **kwargs: Any) -> Any:
 
     @staticmethod
     def get_state() -> Any:
+        try:
+            from hydra.core.plugins import Plugins
+
+            del Singleton._instances[Plugins]
+        except KeyError:
+            pass
```

Comment:
You are changing the actual singleton state, don't do that.
Make a copy of the map without the Plugins singleton.

The only reason this did not cause issues is that plugin interaction is currently only happening on startup, but this can change.